.circleci/config.yml

```diff
@@ -514,15 +549,8 @@ workflows:
           <<: *on-integ-branch
           <<: *after-linter
       - build-and-test-gpu:
-          <<: *on-any-branch
+          <<: *on-integ-branch
```

Comment:
we haven't decided on the design doc where this should be
let's iterate it

Reply:
We'll move this to a github action similar to valgrind - but keep it this way for now, until that is ready.